admin/services/redis-cache.service.js

```diff
@@ -0,0 +1,213 @@
+ * @returns {Boolean}
+ */
+redisCacheService.set = (redisKey, data) => {
+  return new Promise((resolve, reject) => {
+    if (typeof data === 'object') {
+      data = JSON.stringify(data)
+    }
+    redis.set(redisKey, data, () => {
+      console.log(`REDIS (set): Stored \`${redisKey}\``)
+      resolve(true)
+    })
+  })
+}
+
+/**
+ * Called by the sqlService when it performs queries.
+ * If a change query is detected, finds table names and drops any affected caches so they can be redone
+ * @param sql - The SQL query being executed
+ * @returns {Boolean}
+ */
+redisCacheService.dropAffectedCaches = sql => {
+  return new Promise((resolve, reject) => {
+    if (!REDIS_CACHING) {
+      return resolve(true)
+    }
+    let tables = []
+    if (/DELETE|INSERT|UPDATE/.test(sql)) {
+      const tableRegex = /\.\[([a-z]+)\]/gi
+      let match = false
+      while ((match = tableRegex.exec(sql)) !== null) {
+        tables.push(match[1])
+      }
+    }
+    if (/EXEC/.test(sql)) {
+      const spRegex = /\.\[(sp[a-z]+)\]/gi
+      let match = false
+      let procedures = []
+      while ((match = spRegex.exec(sql)) !== null) {
+        procedures.push(match[1])
+      }
+      procedures.forEach(p => {
+        const thisTables = redisCacheService.storedProceduresAffectedTables[p]
+        if (thisTables) {
+          tables = tables.concat(thisTables)
+        }
+      })
+    }
+    if (tables.length) {
+      const stream = redis.scanStream()
+      stream.on('data', keys => {
+        const keyRegex = new RegExp(`(_|-)(${tables.join('|')})`, 'i')
+        const matchedKeys = keys.filter(k => keyRegex.test(k))
+        if (matchedKeys.length) {
+          const pipeline = redis.pipeline()
+          matchedKeys.forEach(key => {
+            console.log(`REDIS (dropAffectedCaches): Dropped \`${key}\``)
+            pipeline.del(key)
+          })
+          pipeline.exec()
+        }
+      })
+      stream.on('end', resolve)
+    } else {
+      resolve(true)
+    }
+  })
+}
+
+/**
+ * Find a Redis key based on a Regex pattern
+ * @param pattern - regex pattern
+ * @returns {Array}
+ */
+const findKeys = pattern => {
+  return new Promise((resolve, reject) => {
+    let foundKeys = []
+    const stream = redis.scanStream()
+    stream.on('data', keys => {
+      foundKeys = foundKeys.concat(keys.filter(key => pattern.test(key)))
+    })
+    stream.on('end', () => {
+      resolve(foundKeys)
+    })
+  })
+}
+
+/**
+ * Manually updates data in Redis and then sends to the Azure `sql-update` queue,
+ * which will be performed in SQL Server by `/functions`
+ * @param serviceKey - the prefix of the key in the format `serviceName.methodName`
+ * @param changes - object with changes in the format { tableName: 'foo-bar', update: { 1: { foo: 'bar' } } }
+ * @returns {Boolean}
+ */
+redisCacheService.update = (serviceKey, changes) => {
+  return new Promise(async (resolve, reject) => {
+    if (!REDIS_CACHING) {
+      resolve(false)
+    } else {
```

Comment:
Please turn the first part of the `if` into a short-circuit and ditch the `else` syntax.